stdlib/public/runtime/Metadata.cpp

```diff
@@ -5547,6 +5542,28 @@ static void recordBacktrace(void *allocation) {
   });
 }
 
+bool swift::_swift_debug_mallocScribbleEnabled = false;
+static void checkMallocScribbleEnvironmentVariable(void *context) {
+  _swift_debug_mallocScribbleEnabled =
+      runtime::environment::SWIFT_DEBUG_ENABLE_MALLOC_SCRIBBLE();
+}
+
+template <typename Pointee>
+static inline void memsetScribble(Pointee *bytes, size_t totalSize) {
+#ifndef NDEBUG
+  // When DEBUG is defined, always scribble.
+  memset(bytes, 0xAA, totalSize);
+#else
+  // When DEBUG is not defined, only scribble when the
+  // SWIFT_DEBUG_ENABLE_MALLOC_SCRIBBLE environment variable is set.
+  static OnceToken_t getenvToken;
+  SWIFT_ONCE_F(getenvToken, checkMallocScribbleEnvironmentVariable, nullptr);
+  if (SWIFT_UNLIKELY(_swift_debug_mallocScribbleEnabled)) {
```

Comment:
The `SWIFT_DEBUG_ENABLE_MALLOC_SCRIBBLE()` call will compile down to a `SWIFT_ONCE_F` and an access to a global variable, so you can remove your own `SWIFT_ONCE_F` and global and just check that directly. (You might have seen me doing this with `_swift_debug_metadataAllocationIterationEnabled`, but that's only because there's additional setup needed when it's enabled.)

Reply:
Thank you for the explanation!  Fixed.